client/tests/e2e/base.py

```diff
@@ -784,3 +800,81 @@ def test_util_for_pod_not_found(self):
                 config=default_configuration, name="foo", namespace="default"
             )
         self.assertIn("Pod foo not found.", str(rte.exception))
+
+
+LOCAL_CONTAINER_NAME = "gefyra-new-backend"
+
+
+@pytest.mark.usefixtures("operator")
+@pytest.mark.usefixtures("tmp_path")
+@pytest.mark.usefixtures("demo_backend_image")
+class GefyraTestCase(TestCase):
+    provider = "k3d"
+
+    @pytest.fixture(autouse=True)
+    def _capture_request(self, request):
+        self._request = request
+
+    def _init_docker(self):
+        self.DOCKER_API = docker.from_env()
+
+    def _init_kube_api(self):
+        load_kube_config(self.kubeconfig)
+        self.K8S_CORE_API = CoreV1Api()
+        self.K8S_RBAC_API = RbacAuthorizationV1Api()
+        self.K8S_APP_API = AppsV1Api()
+        self.K8S_CUSTOM_OBJECT_API = CustomObjectsApi()
+
+    def setUp(self):
+        self.operator: AClusterManager = self._request.getfixturevalue("operator")
+        self.tmp_path = self._request.getfixturevalue("tmp_path")
+        self.demo_backend_image = self._request.getfixturevalue("demo_backend_image")
+        self.kubeconfig = str(self.operator.kubeconfig)
+        self._init_kube_api()
+        self._init_docker()
+        self.operator.apply("tests/fixtures/nginx_exposed.yaml")
+
+    def tearDown(self):
+        containers = ["gefyra-cargo-pytest-gefyra", LOCAL_CONTAINER_NAME]
+        for container in containers:
+            try:
+                self.DOCKER_API.containers.get(container).remove(force=True)
```

Comment:
Note: with that, this test class will evolve to something like "DockerGefyraTestCase" if we cannot abstract over the Docker API as we start supporting Podman.